lib/mix/tasks/zwave.gen.command.ex

```diff
@@ -22,25 +22,124 @@ defmodule Mix.Tasks.Zwave.Gen.Command do
   @switches [no_save: :boolean, target_dir: :string, no_test: :boolean]
 
   def run(args) do
-    {opts, [command_name, command_class | params], _} =
+    {opts, args, _} =
       OptionParser.parse(args, switches: @switches)
 
+    command_name = args |> Enum.at(0) |> maybe_prompt("Command:")
+    command_id = args |> Enum.at(1) |> maybe_prompt_for_command_id()
+    command_class = args |> Enum.at(2) |> maybe_prompt("Command Class:")
+
+    params =
+      args
+      |> Enum.drop(3)
+      |> Enum.reverse()
+      |> Enum.map(&{&1, "any()"})
+      |> prompt_for_params(true)
+      |> Enum.reverse()
+
     target_defs = build_target_defs(command_name, opts)
 
-    copy_files(target_defs, command_name, command_class, params, opts)
+    copy_files(target_defs, command_name, command_id, command_class, params, opts)
   end
 
-  defp copy_files(target_defs, command_name, command_class, params, opts) do
+  defp maybe_prompt(value, prompt) do
+    value = value || String.trim(Mix.shell().prompt(prompt))
+
+    if value == "" or not is_binary(value) do
+      maybe_prompt("", prompt)
+    else
+      value
+    end
+  end
+
+  defp maybe_prompt_for_command_id(id \\ nil) do
+    id = id || String.trim(Mix.shell().prompt("Command ID (hex):"))
+
+    case Regex.run(~r/^0?x?([0-9A-Fa-f]{2})$/, id) do
+      [_, id] -> "0x" <> String.upcase(id)
+      _ -> maybe_prompt_for_command_id()
+    end
+  end
+
+  defp print_param_help() do
+    import IO.ANSI
+
+    Mix.shell().info("""
+    \nEnter command parameters or one of the following commands:
+
+    * #{bright()}a <name> [<type>]#{reset()} - add a parameter
+    * #{bright()}d <name>#{reset()} - delete a parameter by name
+    * #{bright()}v#{reset()} - show parameter list
+    * #{bright()}x#{reset()} - done entering parameters
+    * #{bright()}?#{reset()} - display this message again
+    """)
+  end
+
+  defp print_params(params) do
+    import IO.ANSI
+
+    params
+    |> Enum.reverse()
+    |> Enum.each(fn {n, t} ->
+      Mix.shell().info("* #{green()}#{bright()}:#{n}#{reset()} :: #{yellow()}#{t}#{reset()}")
```

Comment:
```suggestion
      Mix.shell().info(["* ", :green, :bright, ":#{n}", :reset, " :: ", :yellow, "#{t}", :reset])
```

I like this way since it works when redirecting output to a file and ANSI color isn't desirable. However, I see your big multiline string above that will be totally messed up by converting to ansidata. :(

Reply:
The multiline string was easy enough to fix too. Thanks for the suggestion!